data/World/Ganons Castle.json

```diff
@@ -78,7 +87,27 @@
                 Hover_Boots or can_play(Song_of_Time)",
             "Ganons Castle Shadow Trial Golden Gauntlets Chest": "
                 can_use(Fire_Arrows) or 
-                (can_use(Longshot) and (Hover_Boots or can_use(Dins_Fire)))"
+                (can_use(Longshot) and (Hover_Boots or can_use(Dins_Fire)))",
+            "Ganons Castle Shadow Trial Recovery Heart 1": 
+                "(logic_lens_castle or can_use(Lens_of_Truth)) and (
+                    can_use(Fire_Arrows) or 
+                    (can_use(Longshot) and (Hover_Boots or can_use(Dins_Fire))))",
+            "Ganons Castle Shadow Trial Recovery Heart 2": 
+                    "(logic_lens_castle or can_use(Lens_of_Truth)) and (
+                    can_use(Fire_Arrows) or 
+                    (can_use(Longshot) and (Hover_Boots or can_use(Dins_Fire))))",
+            "Ganons Castle Shadow Trial Recovery Heart 3": 
+                    "(logic_lens_castle or can_use(Lens_of_Truth)) and (
+                    can_use(Fire_Arrows) or 
+                    (can_use(Longshot) and (Hover_Boots or can_use(Dins_Fire))))",
+            "Ganons Castle Shadow Trial Like Like Pot 1" : "can_use(Fire_Arrows) or can_use(Longshot)",
+            "Ganons Castle Shadow Trial Like Like Pot 2" : "can_use(Fire_Arrows) or can_use(Longshot)",
+            "Ganons Castle Shadow Trial Pot 1" : "can_use(Megaton_Hammer) and 
+                ((Fire_Arrows and (logic_lens_castle or can_use(Lens_of_Truth))) or 
+                (can_use(Longshot) and (Hover_Boots or (Dins_Fire and (logic_lens_castle or can_use(Lens_of_Truth))))))",
+            "Ganons Castle Shadow Trial Pot 2" : "can_use(Megaton_Hammer) and 
+                ((Fire_Arrows and (logic_lens_castle or can_use(Lens_of_Truth))) or 
+                (can_use(Longshot) and (Hover_Boots or (Dins_Fire and (logic_lens_castle or can_use(Lens_of_Truth))))))"
```

Comment:
You need to wrap these Fire Arrow and Din's Fire checks for the pots at the end of the trial using can_use().

Reply:
You only made this fix on one of the two pots at the end of Shadow Trial.
(You don't need the can_use on Hovers, it doesn't really matter, but the reason you need it on FAs or Dins is b/c that's where the bow and magic checks are.)